code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -712,16 +712,17 @@
 /mob/living/simple_animal/proc/sentience_act() //Called when a simple animal gains sentience via gold slime potion
 	return
 
-/mob/living/simple_animal/update_sight()
+/mob/living/simple_animal/update_sight(reset_sight = FALSE)
 	if(!client)
 		return
 	if(stat == DEAD)
 		grant_death_vision()
 		return
 
-	see_invisible = initial(see_invisible)
-	see_in_dark = initial(see_in_dark)
-	sight = initial(sight)
+	if(reset_sight)

```

Comment:
I think it's because I had to split the vision rework over like 5 PRs due to its sheer size and I'm not sure what went where